content/en/blog/test-coverage/index.md

```diff
@@ -0,0 +1,141 @@
+---
+date: 2023-11-16
+title: "Test Coverage Beyond Unit Testing"
+linkTitle: "Coverage for E2E Tests"
+description: "Automated testing is like having an extra member in your team, a knight against regression. Test coverage measurements and automatic test coverage monitoring help the team keep the knight shiny, i.e., automated test sets in good shape. Traditionally, we have measured the coverage for unit tests, but thorough CI pipelines also include other types of testing. Go has recently introduced new tooling that allows us to measure the test coverage for application tests and thus improve our capabilities for keeping our automated tests in shape automatically."
+author: Laura Vuorenoja
+resources:
+- src: "**.{png,jpg}"
+  title: "Image #:counter"
+---
+
+
+Automated testing is your superpower against regression. It prevents you from breaking existing
+functionality when introducing new code changes. The most important aspect of automated testing
+is **automation**. The testing should happen automatically in the continuous integration (CI) pipeline
+whenever developers create pull requests. If the tests are not passing, the team doesn't merge
+the changes to the main branch. I often regard the automated test set as
+an extra team member – the team's tireless knight against regression.
+
+{{< imgproc cover Fit "925x925" >}}
+<em>Automated test set - the team's knight against regression</em>
+{{< /imgproc >}}
+
+## Complementing Unit Tests
+
+The automated test set usually consists of many types of testing. Unit tests are
+the bread and butter of any project's quality assurance process. But quite often,
+we need something more to complement the unit tests.
+
+We usually write simulation code for external functionality to keep the unit tests relevant,
+modular, and fast to execute. However, wasting the team's resources on creating
+the simulation code doesn't always make sense. Furthermore, we don't include simulation code
+in the end product. Therefore, we should also have testing that verifies the functionality
+with the proper external dependencies.
+
+{{< imgproc application Fit "925x925" >}}
+<em>Findy Agency application tests are run in GitHub Actions workflows and they utilize
```

Comment:
language stuff -> Findy Agency application tests are run in GitHub Actions workflows, utilizing docker-compose to orchestrate the agency services.